docs/permissions.md

```diff
@@ -0,0 +1,112 @@
+- A 2-of-3 threshold with three devices
+- A 2-of-2 threshold with the other two devices
+
+To make a new 3-of-5 with a new coordinator, you would need to share knowledge of both the 2-of-3 and 2-of-2 keys to the new coordinator.
+
+## Coordinator Awareness of Keys
+
+**Access**: Coordinators should disclose to devices which keys they are AWARE of, using a public key.
+
+Devices will register the coordinator as AWARE of these keys for permission purposes.
+
+Key points:
+
+- Under trusted firmware, devices are less likely to collect information than phone coordinators
+- Keygen results in a new key which the coordinator is AWARE of
+- Once the coordinator reveals its awareness of keys, devices can disclose information about held shares:
+  - For AWARE keys: disclose index and share image
+  - For UNAWARE keys: provide a t-of-n key share and key fingerprint that progresses the coordinator towards becoming AWARE
+    - Coordinator must visit `t` devices to become AWARE
+    - Key names may be provided alongside shares
+
+**Confirmation**: Device _could_ REQUIRE CONFIRMATION before informing the coordinator of its held key fingerprints & t-of-n recovery shares.
+
+### Actions
+
+| Action                 | Access Requirement for Coordinator  | On-Device Confirmation |
+| ---------------------- | ----------------------------------- | ---------------------- |
+| KeyGen Start           | AWARE of any key (or no keys exist) | No                     |
+| KeyGen Finish          | AWARE (or no keys exist^^)          | Yes                    |
+| Share Public Nonces    | AWARE or UNAWARE (limited)          | No                     |
+| Signing                | AWARE of relevant key               | Yes                    |
+| Display Backup         | AWARE of relevant key               | Yes                    |
+| Check Backup           | AWARE                               | Yes (keyboard input)   |
+| Restore Backup         | AWARE or UNAWARE                    | Yes                    |
+| Device Data Wipe       | UNAWARE                             | Yes (CPU countdown)    |
+| Factory Firmware Reset | AWARE of ALL keys                   | Yes (CPU countdown?)   |
+| Delete Shares          | AWARE of relevant key               | Yes (long hold)        |
+| Update Firmware        | AWARE of any key                    | Yes                    |
+| Rename Device          | AWARE of any key                    | Yes                    |
+| Device Configuration   | AWARE of ALL keys                   | Yes                    |
+
+## Action Details
+
+### Nonces
+
+Sharing nonces during coordinator recovery enables two-round signing.
+Though we do not want to allow unknown coordinators to request heaps of nonces under different identities to crash the devices.
+Is this limit on nonces during key restoration enough to prevent storage overflow?
+Perhaps the device could store any pending new-coordinator nonces in a separate location, only storing one at a time.
```

Comment:
Why would the number of nonces requested matter? You don't store the nonces locally on the device.

I don't see why storage exhaustion would be a relevant attack since you can just destroy it.